server/core/basic_cluster.go

```diff
@@ -161,6 +197,49 @@ func (bc *BasicCluster) UpdateStoreStatus(storeID uint64, leaderCount int, regio
 
 const randomRegionMaxRetry = 10
 
+// RandNewRegion returns a random region in new region set.
+func (bc *BasicCluster) RandNewRegion(storeID uint64, ranges []KeyRange, optPending RegionOption, optOther RegionOption, optAll RegionOption) *RegionInfo {
+	bc.RLock()
+	if bc.SelectConf.NewRegionFirst {
```

Comment:
I think we can move this condition outside of this function

Reply:
That leaves the scheduler to do the judging. It does reduce the number of function calls.